File: src/codes/june_challenge/day9/IsSubsequence.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0, j = 0;
        while (i < s.length() && j < t.length()) {
            if (s[i] == t[j]) {
                i++;
                j++;
            } else {
                j++;
            }
        }
        return i == s.length();
    }
};

File: src/codes/june_challenge/day9/IsSubsequence.kt
package codes.june_challenge.day9

class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        if (s.count() > t.count()) return false;
        var i = 0
        for (c in t) {
            if (i < s.count() && s[i] == c) i++;
        }
        return i == t.count()
    }
}